script/tool/lib/src/version_check_command.dart

```diff
@@ -595,13 +570,10 @@ ${indentation}The first version listed in CHANGELOG.md is $fromChangeLog.
       }
     }
 
-    if (!state.hasChangelogChange) {
+    if (!state.hasChangelogChange && state.needsChangelogChange) {
```

Comment:
Could there also be a log warning for when a Changelog change is needed but not a version change, suggesting a NEXT entry in the Changelog? I see handling for NEXT cases, but thought a similar log warning could be helpful in this case.

Reply:
This section (specifically the else block just below this) is for failing in that scenario, so that should already be covered. Or do you mean changing the error message to specifically mention using `NEXT`?

Reply:
Yeah, I just meant mentioning `NEXT` since that is a possibility and I remember being slightly confused about that when I first ended up in this scenario!

Reply:
Good idea, I've added a sentence suggesting adding a NEXT entry (and mentioning the docs to point people in the right direction if they haven't encountered it before) if it's not exempt.